shopify_python/git_utils.py

```diff
@@ -95,3 +98,26 @@ def autopep_files(files, max_line_length):
                               select={'W', 'E'},
                               verbose=0)
     autopep8.fix_multiple_files(files, options, sys.stdout)
+
+
+class _CustomPylintReporter(text.ColorizedTextReporter):
```

Comment:
By using a custom subclassed pylint reporter, we get the best of both worlds. We don't have to intercept `stdout` and parse the text, we get colorized output straight to the screen (no blocking delay to collect everything & then display), and we get the raw `Message` objects directly to test against.